spec/factories/comments.rb

```diff
@@ -1,11 +1,7 @@
 FactoryBot.define do
-  factory :comment do
+  factory :comment, aliases: [:article_comment] do
     user
-    body_markdown      { Faker::Hipster.paragraph(sentence_count: 1) }
-    commentable_id     { rand(1000) }
-    commentable_type   { "Article" }
-    factory :article_comment do
-      association :commentable, factory: :article
-    end
+    body_markdown { Faker::Hipster.paragraph(sentence_count: 1) }
+    association :commentable, factory: :article
```

Comment:
I like this, though I wonder if this doesn't increase the amount of objects created per each test that doesn't explicitly call `create(:comment, commentable: ...)`

Reply:
If `commentable` (or both `commentable_id` and `commentable_type`) was not specified before, creating comment would fail because with a validation error `Commentable must exist`. What was happening before the change: `commentable_id` was specified in the tests when using a factory and the default value was used for the `commentable_type` field. So, the associated records were created anyway.
Now the factory can be used both with or without specifying `commentable`.